### PR TITLE
util: migrate test-infra to testify for dbutil,filter,table-filter

### LINKED ISSUE #33454: migrate test-infra to testify for `util/dbutil` pkg

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #33454 #33456 #33458 #33459

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: util/dbutil/common_test.go [hunk 1]
```diff
@@ -241,6 +241,6 @@ func (s *testDBSuite) TestFormatTimeZoneOffset(c *C) {
 
 	for k, v := range cases {
 		offset := FormatTimeZoneOffset(v)
-		c.Assert(k, Equals, offset)
+	require.Equal(t, offset, k)
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply util/dbutil/common_test.go]: Thank. Fixed.

**Comment by tisonkun** [comment]
@hawkingrei ci failed on migrated tests. PTAL.

```
--- FAIL: TestIsSystemSchema (0.00s)
    schema_test.go:42: 
        	Error Trace:	schema_test.go:42
        	Error:      	Not equal: 
        	            	expected: bool(true)
        	            	actual  : string("schema name = %s")
        	Test:       	TestIsSystemSchema
        	Messages:   	information_schema%!(EXTRA bool=true)
FAIL
```

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for taking over these tasks!

**Comment by tisonkun** [comment]
Shall https://github.com/pingcap/tidb/issues/33459 be closed by this PR also?

**Comment by hawkingrei** [comment]
> Shall #33459 be closed by this PR also?

Yes, I have been updated it in the pr description.

**Review by xhebox — APPROVED** [review]

**Pull request merged**